sims/verilator/Makefile

```diff
@@ -1,112 +1,207 @@
+#############################################################################
+# verilator simulator types and rules
+#############################################################################
 sim_prefix = simulator
 sim = $(sim_dir)/$(sim_prefix)-$(MODEL_PACKAGE)-$(CONFIG)
 sim_debug = $(sim_dir)/$(sim_prefix)-$(MODEL_PACKAGE)-$(CONFIG)-debug
 
+# verilator doesn't use +permissive, but common.mk expects the simulator's
+# Makefile to set these to be consistent with VCS
 PERMISSIVE_ON=
 PERMISSIVE_OFF=
 
+# common.mk, expects the simulator's Makefile to set this
 WAVEFORM_FLAG=-v$(sim_out_name).vcd
 
-.PHONY: default debug
-default: $(sim)
-debug: $(sim_debug)
-
-#########################################################################################
+#############################################################################
 # import other necessary rules and variables
-#########################################################################################
+#############################################################################
 include $(base_dir)/common.mk
 
-#########################################################################################
-# verilator binary and flags
-#########################################################################################
+# declare the compile targets here since sim/sim_debug are defined
+compile: $(sim)
+compile-debug: $(sim_debug)
+
+#############################################################################
+# verilator-specific user-interface variables and commands
+#############################################################################
+HELP_COMPILATION_VARIABLES += \
+"   ENABLE_PRINTF_PATTERN  = python regex. any printf() with text that matches this regex will be " \
+"                            force enabled in sim output default is no selectively enabled printf's" \
+"   VERILATOR_PROFILE      = 'none' if no verilator profiling (default)" \
+"                            'all' if full verilator runtime profiling" \
+"                            'threads' if runtime thread profiling only" \
+"   VERILATOR_THREADS      = how many threads the simulator will use (default 1)" \
+"   VERILATOR_FST_MODE     = enable FST waveform instead of VCD. use with debug build"
+
+HELP_COMMANDS += \
+"   compile|default        = compiles non-debug simulator [./$(shell basename $(sim))]" \
+"   compile-debug|debug    = compiles debug simulator [./$(shell basename $(sim_debug))]" \
+"   clean                  = remove all debug/non-debug simulators and intermediate files" \
+"   clean-sim              = removes non-debug simulator and verilator-generated files" \
+"   clean-sim-debug        = removes debug simulator and verilator-generated files"
+
+#############################################################################
+# selectively enabling printfs
+#############################################################################
+ENABLE_PRINTF_PATTERN ?=
```

Comment:
This is not verilator-specific, so this should be in `variables.mk`

Reply:
correct.